pkg/operator/targetconfigcontroller/targetconfigcontroller.go

```diff
@@ -277,6 +278,16 @@ func managePod(client coreclientv1.ConfigMapsGetter, recorder events.Recorder, o
 	}
 	required.Spec.Containers[0].Args = append(required.Spec.Containers[0].Args, fmt.Sprintf("-v=%d", v))
 
+	var observedConfig map[string]interface{}
+	if err := yaml.Unmarshal(operatorConfig.Spec.ObservedConfig.Raw, &observedConfig); err != nil {
+		return nil, false, fmt.Errorf("failed to unmarshal the observedConfig: %v", err)
+	}
+	proxyConfig, _, err := unstructured.NestedStringMap(observedConfig, "proxy")
+	if err != nil {
+		return nil, false, fmt.Errorf("couldn't get the proxy config from observedConfig: %v", err)
+	}
+	required.Spec.Containers[0].Env = append(required.Spec.Containers[0].Env, proxyMapToEnvVars(proxyConfig)...)
```

Comment:
do we want to check if the proxy var are already set?

Reply:
will the last one win?

The blueprint for this comes from us. The user has no influence on it. So I don't think we need further validation.

Reply:
@sttts if the required.Spec.Containers[0].Env already have the proxy vars from us, is this going to keep adding them over and over? ;-) I think that was my question. Agree, user won't be able to change this (we will stomp).

Reply:
It's a good point, although I'd say we may want to decide that once we have the need to deviate from whatever there is in the `Proxy` object, as at least for the `noProxy` field, that's going to be fed with some cluster defaults from the very start if I understand the design of the feature properly.